File: LeetCode/643/main.py
class Solution:
    def findMaxAverage(self, arr: List[int], k: int) -> float:
        
        n = len(arr)
        left, s, ans = 0, 0, 0

        for i in range(n) :
            s += arr[i]
            ws = i - left + 1
            if(ws > k) :
                s -= arr[left]
                left += 1
            elif(ws == k) :
                ans = max(ans, s / ws)

        return ans            

File: LeetCode/643/main.cpp
class Solution {
public:
  double findMaxAverage(vector<int> &arr, int k) {
    int n = arr.size(), left = 0;
    double s = 0, ans = 0;

    for (int i = 0; i < k; i++) {
      s += arr[i];
    }

    ans = s / k;

    for (int i = k; i < n; i++) {
      s -= arr[left];
      s += arr[i];
      left++;
      ans = max(ans, s / k);
    }

    return ans;
  }
};